packages/matchbox/src/components/Tabs/Tabs.js

```diff
@@ -66,14 +88,87 @@ function Tabs(props) {
     if (onSelect && selected !== index) {
       onSelect(index, selected);
     }
+
+    if (isOverflowing) {
+      setPopoverOpen(false);
+    }
   }
 
+  React.useLayoutEffect(() => {
+    const wrapperPosition = getPositionFor(wrapperRef.current);
+    const overflowPosition = getPositionFor(overflowRef.current);
+
+    // Compares tab container width with its position relative to a measurement pixel
+    if (wrapperPosition.width < overflowPosition.left - wrapperPosition.left) {
+      setIsOverflowing(true);
+    } else {
+      setIsOverflowing(false);
+    }
+  }, [wrapperRef, overflowRef, windowSize]);
+
+  const selectedTab = tabs[selected];
+
+  // Constructs ActionList actions from tabs
+  const tabActions = React.useMemo(() => {
+    return tabs.map((tab, i) => {
+      return { is: 'button', ...tab, onClick: e => handleClick(e, i), visible: i !== selected };
+    });
+  }, [selected, tabs, isOverflowing]);
+
   return (
-    <StyledTabs {...pick(rest)}>
-      {tabs.map((tab, i) => (
-        <Tab key={i} index={i} fitted={fitted} selected={selected} {...tab} onClick={handleClick} />
-      ))}
-    </StyledTabs>
+    <Container borderBottom="400" {...pick(rest)} ref={wrapperRef}>
+      <Box aria-hidden={isOverflowing} overflow="hidden">
+        <OverflowTabContainer
+          aria-orientation="horizontal"
+          isOverflowing={isOverflowing}
+          role="tablist"
+        >
+          {tabs.map((tab, i) => (
+            <Tab
+              key={i}
+              index={i}
+              fitted={fitted}
+              selected={selected}
+              {...tab}
+              onClick={handleClick}
+            />
+          ))}
+          {/* Measurement pixel used to detect content overflow */}
+          <Box display="inline-block" width="1px" height="1px" ref={overflowRef} />
+        </OverflowTabContainer>
+      </Box>
+      {isOverflowing && (
+        <Box display="flex" alignItems="center" justifyContent="space-between">
+          <Box flex="0">
+            <Tab index={selected} selected={selected} {...selectedTab} />
+          </Box>
+
+          <Box flex="0">
+            <Popover
+              id="tab-options"
+              left
+              bottom
+              open={popoverOpen}
+              onClose={() => setPopoverOpen(false)}
+              trigger={
+                <Button
+                  aria-controls="tab-options"
+                  aria-expanded={popoverOpen}
+                  data-id="tab-options-button"
+                  flat
+                  onClick={() => setPopoverOpen(true)}
```

Comment:
Not necessary, but might improve UX to toggle open and closed when clicking this button. As it is right now I can click to open but have to click off of the button to close.

Reply:
Hmm yea good catch